aws/src/test/java/org/apache/iceberg/aws/glue/InMemoryLockManagerTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InMemoryLockManagerTest {
+
+  private LockManagers.InMemoryLockManager lockManager;
+  private String lockEntityId;
+  private String ownerId;
+
+  @Before
+  public void before() {
+    lockEntityId = UUID.randomUUID().toString();
+    ownerId = UUID.randomUUID().toString();
+    lockManager = new LockManagers.InMemoryLockManager(Maps.newHashMap());
```

Comment:
How long does this take to run? It seems like a good idea to use some settings to avoid long runtimes. What about setting the timeout to 1s or something?

Reply:
Added a timeout rule

Reply:
Sorry, I meant configuring the lock manager timeouts to small increments, not adding a Timeout. Since you adjust settings in individual tests, it should be fine.